src/LinearSolvers/preconditioners.jl

```diff
@@ -288,16 +288,16 @@ Update the preconditioner `p` from the sparse Jacobian `J` in CSC format for the
 Note that this implements the same algorithm as for the GPU and becomes very slow on CPU with growing number of blocks.
 
 """
-function update(J::SparseMatrixCSC, p)
-    kernel! = update_cpu_kernel!(CPU())
+function update(p, J::SparseMatrixCSC, device)
+    kernel! = update_cpu_kernel!(device)
     p.P.nzval .= 0.0
-    ev = kernel!(J.colptr, J.rowval, J.nzval, p, p.lpartitions, ndrange=p.nblocks, dependencies=Event(CPU()))
+    ev = kernel!(J.colptr, J.rowval, J.nzval, p, p.lpartitions, ndrange=p.nblocks, dependencies=Event(device))
     wait(ev)
     return p.P
 end
-function update(J::Transpose{T, SparseMatrixCSC{T, I}}, p) where {T, I}
+function update(p, J::Transpose{T, SparseMatrixCSC{T, I}}) where {T, I}
```

Comment:
Not sure this function is tested in the tests.

Reply:
I don't even know what it is used for :-) . Shouldn't we add a test then?